### DISCUSSION TITLE
How can I debug planning engine C++ code in Visual Studio Code?

### DISCUSSION THREAD

**Discussion by qizhengzhong** [frePPLe/frepple#658] [Q&A]
Hi, 
Does anyone know how to debug the C++ part of the frepple code in VSC? 
I have successfully start up the "Planning engine Linux" that was configured [here](https://github.com/frePPLe/frepple/blob/master/.vscode/launch.json) in VSC. 

But how do I trigger the debugging point that I triggered in the "forcastsolver.cpp" for example? 
I tried running the execute plan step through frepple UI, but it didn't trigger anything. 

Thanks in advance!

**Comment by jdetaeye** [comment]
The "planning engine linux" debugging topic is exactly what you need.
It launches the planning engine with the very same command line and its arguments as the UI does.  

![image](https://github.com/frePPLe/frepple/assets/5219207/d1b355d9-9663-46f6-b68d-c3a62abb2af7)
